### DISCUSSION TITLE
Temperature of CPU node

### DISCUSSION THREAD

**Discussion by anarro** [dougiteixeira/proxmoxve#117] [Feature request] [answered]
It's possible add the temperature off node?

**Comment by dougiteixeira** [comment]
The Proxmox API does not have this feature available.  So it is not possible to include.

**Reply by WholeSon** [reply]
Is it possible to somehow pull up `lm-sensors`?

**Reply by Stephan-4711** [reply]
I installed the lm-sensors too. It would really be great if this could be integrated to monitor the temperature of the system

**Reply by alexruffell** [reply]
Maybe Proxmox takes feature requests... It indeed would be nice to monitor hardware health of the machine from Proxmox itself so they may be open to it.

**Reply by alexruffell** [reply]
Is lm-sensors supported? Based on https://github.com/lm-sensors/lm-sensors it would appear dead... or maybe it is something that doesn't need updating in years?

**Comment by dougiteixeira** [comment]
The Proxmox API does not have this feature available. So it is not possible to include.

However, you can use an alternative similar to [the script for sending CPU temperature](https://github.com/dougiteixeira/proxmoxve/blob/main/scripts_cron/cpu_temperature.md).

Good luck!

**Comment by Gugulator** [comment]
may be it is possible to use this command to get cpu temperature info?
`cat /sys/class/hwmon/hwmon1/temp1_input | awk '{print substr($0, 1, length($0)-3)}'`
not need to install lm-sensors

**Comment by dougiteixeira** [comment, marked as answer]
The function of this integration is to consume the data provided by the Proxmox API, so there is no way to include information from other sources (for example, a CLI command). There are alternatives for this.

WHEN and IF this feature is integrated into Proxmox it will be here, but don't get excited about it, [here is the request for this feature](https://bugzilla.proxmox.com/show_bug.cgi?id=208).

**Reply by Gugulator** [reply]
yeah, i saw it. case opened from 2012 and still no official solution

**Reply by FRHOME** [reply]
maybe a clue here ? 
https://github.com/Meliox/PVE-mods

**Comment by mZ738** [comment]
Hardware readings are in since 5.0.0: temperatures, fan speeds, voltages and power from `lm-sensors`, through [PVE-mods](https://github.com/Meliox/PVE-mods) on the node — the integration reads them from the node status the mod extends, no extra API calls. Both the current `node_info` package and the legacy script are recognised. How to set it up, which chips are classified how, and why readings come and go with the mod's on-demand collector: [Hardware sensors](https://github.com/dougiteixeira/proxmoxve/blob/main/docs/hardware-sensors.md). 5.1.0 ([release notes](https://github.com/dougiteixeira/proxmoxve/releases/tag/5.1.0)) is out today; I am closing this as done — please open a new thread if a chip of yours is not picked up.
